vcpkg/consume/install-locally-modified-package.md

```diff
@@ -0,0 +1,251 @@
+---
+
+# Tutorial: Install a locally modified dependency
+This tutorial guides you on how to modified a vcpkg port using custom overlays.
+We recommend that you read the tutorial on [packaging a
+library](../get_started/get-started-packaging.md) before proceeding.
+
+In this tutorial, you will learn to:
+
+> [!div class="checklist"]
+> * [Create an overlay port](#1---create-an-overlay-port)
+> * [Get the port's source code](#2---get-the-ports-source-code)
+> * [Create a temporary Git registry](#3---create-a-temporary-git-registry)
+> * [Modify the necessary files](#4---modify-the-necessary-files)
+> * [Generate a patch file](#5---generate-a-patch-file)
+> * [Modify `portfile.cmake` to apply the patch file](#6---modify-portfilecmake-to-apply-the-patch)
+> * [Install your overlay port](#7---install-your-overlay-port)
+
+## Prerequisites
+
+* A terminal
+* vcpkg
+* [Git](<https://git-scm.com/downloads>)
+* A C++ compiler
+
+## 1 - Create an overlay port
+The first step is to create an [overlay port](../concepts/overlay-ports.md) of
+the package you want to modify.
+
+1 - Create a directory to hold your overlay ports
+
+The overlay ports directory can be created in any filesystem location of your
+choosing. In any step of this tutorial replace `$OVERLAY_LOCATION` with your
+chosen location.
+
+::: zone pivot="shell-powershell, shell-cmd"
+
+```powershell
+mkdir "$OVERLAY_LOCATION"
+```
+
+::: zone-end
+::: zone pivot="shell-bash"
+
+```bash
+mkdir "$OVERLAY_LOCATION"
+```
+
+::: zone-end
+
+2 - Copy the contents of the port into your overlay ports directory
+
+For this tutorial, you'll modify the `vcpkg-sample-library` port in the
+[packaging tutorial](../get_started/get-started-packaging.md) to add dynamic
+library support.
+
+::: zone pivot="shell-powershell"
+
+```powershell
+Copy-Item -Path <path/to/vcpkg-sample-library> -Destination "$OVERLAY_LOCATION" -Recurse
+```
+
+::: zone-end
+::: zone pivot="shell-cmd"
+
+```console
+xcopy <path/to/vcpkg-sample-library> "$OVERLAY_LOCATION" /E
+```
+
+::: zone-end
+::: zone pivot="shell-bash"
+
+```bash
+cp -R <path/to/vcpkg-sample-library> "$OVERLAY_LOCATION"
+```
+::: zone-end
+
+## 2 - Get the port's source code
+To obtain the source code for the port you want to modify, run the following command:
+
+`vcpkg install "--overlay-ports=$OVERLAY_LOCATION" vcpkg-sample-library --editable`
+
+The output should contain a line similar to:
+
+```Console
+-- Using source at path/to/vcpkg/buildtrees/vcpkg-sample-library/src/1.0.0-b54c55c215
+```
+
+This is the location of the port's source code.
+
+## 3 - Create a temporary Git registry
+Create a temporary Git registry in the source code location, this is in  order
+to use Git to generate patch files that you can apply in the `portfile.cmake`
+script. Replace `$SOURCE_PATH` with the location obtained in the previous step.
+
+```bash
+cd "$SOURCE_PATH"
+git init
+git add .
+git commit -m "Initial commit"
+```
```

Comment:
This section would apply to cmd and PowerShell too, so maybe use the "console" type rather than "bash"?